packages/camera/lib/camera.dart

```diff
@@ -0,0 +1,278 @@
+        name: camera['name'],
+        lensDirection: _parseCameraLensDirection(camera['lensFacing']),
+      );
+    }).toList();
+  } on PlatformException catch (e) {
+    throw new CameraException(e.code, e.message);
+  }
+}
+
+class CameraDescription {
+  final String name;
+  final CameraLensDirection lensDirection;
+  CameraDescription({this.name, this.lensDirection});
+
+  @override
+  bool operator ==(Object o) {
+    return o is CameraDescription &&
+        o.name == name &&
+        o.lensDirection == lensDirection;
+  }
+
+  @override
+  int get hashCode {
+    return hashValues(name, lensDirection);
+  }
+
+  @override
+  String toString() {
+    return '$runtimeType($name, $lensDirection)';
+  }
+}
+
+class CameraException implements Exception {
+  String code;
+  String description;
+  CameraException(this.code, this.description);
+
+  @override
+  String toString() => '$runtimeType($code, $description)';
+}
+
+class CameraPreview extends StatelessWidget {
+  final CameraController controller;
+  const CameraPreview(this.controller);
+
+  @override
+  Widget build(BuildContext context) {
+    return controller.value.initialized
+        ? new Texture(textureId: controller._textureId)
+        : new Container();
+  }
+}
+
+class CameraValue {
+  /// True if the camera is on.
+  final bool isStarted;
+  final bool initialized;
+  final String errorDescription;
+
+  const CameraValue({this.isStarted, this.initialized, this.errorDescription});
+
+  const CameraValue.uninitialized()
+      : this(isStarted: true, initialized: false);
+
+  bool get isErroneous => errorDescription != null;
+
+  CameraValue copyWith({
+    bool isStarted,
+    bool initialized,
+    String errorDescription,
+  }) {
+    return new CameraValue(
+      isStarted: isStarted ?? this.isStarted,
+      initialized: initialized ?? this.initialized,
+      errorDescription: errorDescription ?? this.errorDescription,
+    );
+  }
+
+  @override
+  String toString() {
+    return '$runtimeType('
+        'started: $isStarted, '
+        'initialized: $initialized, '
+        'errorDescription: $errorDescription)';
+  }
+}
+
+/// Controls a device camera.
+///
+/// Use [availableCameras] to get a list of available cameras.
+///
+/// Before using a [CameraController] a call to [initialize] must complete.
+///
+/// To show the camera preview on the screen use a [CameraPreview] widget.
+class CameraController extends ValueNotifier<CameraValue> {
+  final CameraDescription description;
+  final ResolutionPreset resolutionPreset;
+  int _textureId;
+  bool _disposed = false;
+  StreamSubscription<Map<String, dynamic>> _eventSubscription;
+  Completer<Null> _creatingCompleter;
+  Size _previewSize;
+
+  CameraController(this.description, this.resolutionPreset)
+      : super(const CameraValue.uninitialized());
+
+  /// The size of the preview in pixels.
+  ///
+  /// Is [null] until [initialize] completes.
+  Size get previewSize => _previewSize;
```

Comment:
So client code would be `(await controller.initialize()).previewSize`?

Just wondering if it would be simpler on the client to move `previewSize` and `aspectRatio` to the `CameraValue` with reasonable/custom defaults.